src/amo/withExperiment.js

```diff
@@ -302,6 +312,25 @@ export const withExperiment =
         return Object.keys(registeredExperiments).includes(id);
       }
 
+      // Warning: Please think carefully before using this function. It was
+      // introduced to address a specific requirement for an experiment to
+      // avoid having to store data in a cookie, but realistically one should
+      // never change the variant for a user after they have been enrolled in
+      // an experiment.
+      // See https://github.com/mozilla/addons-frontend/issues/10770
+      updateVariant = (variant: string) => {
```

Comment:
I'd prefer we don't introduce any of this in this HOC for the reasons stated in the code comment above. It is hack and it shouldn't be exposed on the public API of a HOC that could be used for other more sane things.

Reply:
I added the copious comments in an attempt to avoid this being misused. The reason I put it here is because, although it's an ugly hack, until we come up with a better way of dealing with this use case, we're probably going to have to keep doing this. Also, ultimately this is where it belongs, as otherwise we are leaking the internals of `withExperiment` to other components and that seems like an equally bad thing. So, I'd actually prefer to keep it here, along with the warning in the comments, but if you feel strongly that it should be removed and replaced with your suggestion at https://github.com/mozilla/addons-frontend/pull/10787#discussion_r675506854, I will do so. Let me know what you think.

Reply:
> Let me know what you think.

I'd prefer to not update this HOC.